scripts/Connector/Connector_JST/conn_jst_XA_vertical.py

```diff
@@ -0,0 +1,271 @@
+#!/usr/bin/env python3
+
+import sys
+import os
+#sys.path.append(os.path.join(sys.path[0],"..","..","kicad_mod")) # load kicad_mod path
+
+
+# export PYTHONPATH="${PYTHONPATH}<path to kicad-footprint-generator directory>"
+sys.path.append(os.path.join(sys.path[0], "..", "..", ".."))  # load parent path of KicadModTree
+import argparse
+import yaml
+from helpers import *
+from KicadModTree import *
+
+sys.path.append(os.path.join(sys.path[0], "..", "..", "tools"))  # load parent path of tools
+from footprint_text_fields import addTextFields
+
+series = "XA"
+manufacturer = 'JST'
+orientation = 'V'
+number_of_rows = 1
+datasheet = 'http://www.jst-mfg.com/product/pdf/eng/eXA1.pdf'
+
+#Note : this script should cover both standard and "N type" variants. Apparently the N type only uses a different platic:
+# https://jsttac.blogspot.com/2011/08/what-is-difference-in-xa-series-header.html
+
+fab_pin1_marker_type = 1
+pin1_marker_offset = 0.3
+pin1_marker_linelen = 1.25
+
+drill_size = 0.95 #Datasheet: 0.9 +0.1/-0
+mh_drill = 1.25 #for boss
+pad_to_pad_clearance = 0.8
+pad_copper_y_solder_length = 0.5 #How much copper should be in y direction?
+min_annular_ring = 0.15
+
+
+
+pitch = 2.5
+
+# as of 2020 : available in 2- to 15-pin, 18 and 20-pin variants.
```

Comment:
Please check this since the latest datasheet doesn't show 18 pin. I don't see it as of 30 Sep, anyway.